File: Design-Patterns/Bridge/CSharp/Bridge/Characters/Orc.cs
namespace Bridge.Characters
{
    public class Orc : Character
    {
        public Orc(int id, string name, int health, int mana) : base(id, name, health, mana)
        {

        }

        public override void TakeDamage(int value)
        {
            value -= 3;
            if (value <= 0) value = 0;

            base.TakeDamage(value);
        }
    }
}

File: Design-Patterns/Bridge/Cpp/orc.cpp
#include "character.h"

orc::orc(int id, std::string name, int health, int mana) : character(id, name, health, mana)
{
	
}

void orc::take_damage(int value)
{
	value -= 3;
	if (value <= 0) value = 0;

	orc::take_damage(value);
}
